### PR TITLE
update to use the newer externpro

### PR DESCRIPTION
**Pull request by swflb**

### CODE REVIEW

File: Findexternpro.cmake [hunk 1]
```diff
@@ -84,13 +91,13 @@ find_path(externpro_DIR
...
-    "$ENV{externpro_DIR}/externpro-${externpro_SIG}-${CMAKE_SYSTEM_NAME}"
+    "$ENV{extern_DIR}/externpro ${externpro_SIG}"
+    "$ENV{extern_DIR}/externpro-${externpro_SIG}-${CMAKE_SYSTEM_NAME}"
     # installed versions
     "$ENV{ProgramW6432}/externpro ${externpro_SIG}"
     "$ENV{${PFX86}}/externpro ${externpro_SIG}"
-    "~/externpro/externpro-${externpro_SIG}-${CMAKE_SYSTEM_NAME}"
-    "/opt/externpro/externpro-${externpro_SIG}-${CMAKE_SYSTEM_NAME}"
+    "~/extern/externpro-${externpro_SIG}-${CMAKE_SYSTEM_NAME}"
+    "/opt/extern/externpro-${externpro_SIG}-${CMAKE_SYSTEM_NAME}"
```
Comment (brobeson) [inline Findexternpro.cmake:100]: Why these changes? And those on lines 94 and 95?

File: projects/README.md [hunk 2]
```diff
@@ -2,16 +2,15 @@
...
+|[HDF-GROUP](http://www.hdfgroup.org 'The HDF Group')|[open](https://www.hdfgroup.org/ftp/HDF5/current/src/unpacked/COPYING 'HDF5 License')|HDF5 is a data model, library, and file format for storing and managing data|1.8.16|none|none|
+|[librdkafka](https://github.com/edenhill/librdkafka 'librdkafka on github')|[open](https://github.com/edenhill/librdkafka/blob/master/LICENSE '2-clause BSD license')|librdkafka is a C library implementation of the Apache Kafka protocol, containing both Producer and Consumer support|v0.9.5|[repo](https://github.com/distributePro/librdkafka 'distributePro fork of librdkafka repo on github')|[diff](https://github.com/distributePro/librdkafka/compare/edenhill:v0.9.5...xp-v0.9.5 'patch/diff')|
+|[NOVAS](http://aa.usno.navy.mil/software/novas/novas_info.php 'NOVAS')|[open](http://aa.usno.navy.mil/software/novas/novas_c/README.txt '(See Section IV. Using NOVAS in Your Applications)')|NOVAS is an integrated package of ANSI C functions for computing many commonly needed quantities in positional astronomy.|3.1|none|none|
```
Comment (brobeson) [inline projects/README.md]: Why was Kerberos removed?

File: projects/README.md [hunk 3]
```diff
@@ -2,16 +2,15 @@
...
+|[Qt](http://qt.io/ 'Qt - Home')|[LGPL](http://www.qt.io/qt-licensing-terms/ 'LGPL')|One Qt Code: Create Powerful Applications & Devices|5.8.0|[repo](http://code.qt.io/cgit/qt/qt5.git 'Qt5 main repo')|none|
+|[Qwt](http://http://qwt.sourceforge.net/ 'Qwt - Qt Widgets for Technical Applications')|[LGPL](http://qwt.sourceforge.net/qwtlicense.html 'LGPL with exceptions')|The Qwt library contains GUI Components and utility classes which are primarily useful for programs with a technical background.|6.1.2|none|none|
+|[ZEROMQ](https://zeromq.org 'ZEROMQ')|[open](https://github.com/zeromq/libzmq/blob/master/README.md '(See License Section. GNU Lesser General Public LIcense.)')|ZEROMQ is a lightweight messaging kernel is a library that extends standard socket interfaces.|4.2.2|none|none|
+|[Zookeeper](https://zookeeper.apache.org/ 'Zookeeper - Home')|[open](http://www.apache.org/licenses/ 'Apache V2.0')|Apache ZooKeeper is an effort to develop and maintain an open-source server which enables highly reliable distributed coordination.|3.4.9|[repo](https://github.com/apache/zookeeper 'Zookeeper main repo')|none|
```
Comment (brobeson) [inline projects/README.md]: Why was cppzmq removed?

**Review by brobeson — COMMENTED** [review]
What are the _ip_ functions?

File: Findexternpro.cmake (hunk 1 above)
Comment (swflb) [reply Findexternpro.cmake:100]: @brobeson  this file is a copy from externpro.

File: projects/README.md (hunk 2 above)
Comment (swflb) [reply projects/README.md]: @brobeson for some reason when I run cmake, this file changes. I don't know why. This is the results

Comment (brobeson) [reply projects/README.md]: Did you disable building Kerberos and cppzmq? I _think_, the readme is updated as part of the build process, so if something isn't built, it's removed.

**Review by brobeson — APPROVED** [review]

File: projects/README.md (hunk 2 above)
Comment (bnhhughes) [reply projects/README.md]: This README will get regenerated at build time - but I've avoided committing the one that is produced just because on Linux we don't build some of the pieces that are built on Windows and it will leave them out of this file.  There are also a couple of diffs that are input manually for some of the projects and those will also be removed if you commit this file...

File: projects/zookeeper.cmake [hunk 4]
```diff
@@ -23,7 +23,7 @@ set(PRO_ZOOKEEPER
   GIT_ORIGIN ${ZK_REPO}
   GIT_TAG release-${ZK_VER}
   DLURL ${ZK_REPO}/archive/release-${ZK_VER}.tar.gz
-  DLMD5 23df0e0bccf5d05d8190fd8ef459947c
+  DLMD5 790c9b028f2f9c6ed17938a396365b74
```
Comment (bnhhughes) [inline projects/zookeeper.cmake:26]: How did the hash change and not the version?

**Comment by bnhhughes** [comment]
Just a general question - was there a compelling reason to update to the new externpro or was this just so we're not so far behind?  I don't think there was anything new added to it that necessarily benefits ovsrpro, though I'm not opposed to updating.

File: projects/README.md [hunk 5]
```diff
@@ -2,16 +2,15 @@
 
 |project|license|description|version|repository|patch/diff|
 |-------|-------|-----------|-------|----------|----------|
+|[CPPZMQ](https://github.com/zeromq/cppzmq 'CPPZMQ')|[MIT](https://github.com/zeromq/cppzmq/blob/v4.2.1/LICENSE 'MIT')|CPPZMQ is a minimal c++ binding to the libzmq functions.|4.2.1|none|none|
 |[FFmpeg](https://www.ffmpeg.org/ 'FFmpeg website')|[LGPL](https://www.ffmpeg.org/legal.html 'Lesser GPL v2.1')|complete, cross-platform solution to record, convert and stream audio and video|2.6.2|[repo](https://github.com/distributepro/FFmpeg 'forked FFmpeg repo on github')|[diff](https://github.com/distributepro/FFmpeg/compare/FFmpeg:n2.6.2...xp2.6.2 'patch/diff')|
```
Comment (bnhhughes) [inline projects/README.md]: I would say this is a good change to move CPPZMQ here to the beginning so it's alphabetical but I think you should probably revert the rest of the changes to this file, mostly it looks to be removing the manually added diff portions.

File: projects/README.md [hunk 6]
```diff
@@ -2,16 +2,15 @@
...
-|[Qt](http://qt.io/ 'Qt - Home')|[LGPL](http://www.qt.io/qt-licensing-terms/ 'LGPL')|One Qt Code: Create Powerful Applications & Devices|5.8.0|[repo](http://code.qt.io/cgit/qt/qt5.git 'Qt5 main repo')|[diff](https://github.com/distributePro/ovsrpro/blob/master/patches/qt5.patch 'patch/diff')|
-|[Qwt](http://http://qwt.sourceforge.net/ 'Qwt - Qt Widgets for Technical Applications')|[LGPL](http://qwt.sourceforge.net/qwtlicense.html 'LGPL with exceptions')|The Qwt library contains GUI Components and utility classes which are primarily useful for programs with a technical background.|6.1.2|none|[diff](https://github.com/distributePro/ovsrpro/blob/master/patches/qwtconfig.pri.patch 'patch/diff')|
-|[ZeroMQ](https://zeromq.org/ 'ZeroMQ - Home")|[LGPL](https://github.com/zeromq/libzmq/blob/master/README.md/ 'LGPL v3')|ZeroMQ is a lightweight messaging kernel that extends standard socket interfaces with more powerful features.|4.2.2|[repo](https://github.com/zeromq/libzmq.git)|none|
```
Comment (bnhhughes) [inline projects/README.md]: You could also fix this entry - change the `"` after Home to a `'` and it should fix the issue with it not displaying properly (I believe that will resolve issue #27)

File: projects/zookeeper.cmake (hunk 4 above)
Comment (swflb) [reply projects/zookeeper.cmake:26]: @bnhhughes it changed on the server side. They must have posted a new file

Comment (bnhhughes) [reply projects/zookeeper.cmake:26]: Hmm, that's weird...

**Review by bnhhughes — APPROVED** [review]
You could go ahead and make the simple fix in the README that I mentioned for ZeroMQ if you want.

**Comment by brobeson** [comment]
@bnhhughes 

> was there a compelling reason to update to the new externpro

The short answer is so @swflb and I can use GCC 7.2. It's a complicated mess because Gnu changed how GCC reports its version.

**Comment by smanders** [comment]
externpro 18.01.1 https://github.com/smanders/externpro/releases/tag/18.01.1 also has FFmpeg 2.6.2, OpenH264 1.4.0, and builds without needing to install yasm (because it builds it and uses it internally) -- so I would say those are compelling reasons to update to the new externpro and certainly are new things added that would benefit ovsrpro (these projects could/should be removed from ovsrpro, IMHO)

File: projects/README.md (hunk 2 above)
Comment (smanders) [reply projects/README.md]: the README shouldn't change based on what platform you are on -- there are simply *many* issues with how the ovrsrpro projects have been done...

File: Findexternpro.cmake (hunk 1 above)
Comment (smanders) [reply Findexternpro.cmake:100]: this is now the default install location on unix systems: `/opt/extern` and `~/extern` because there are other projects (for SkyLynx development) that are using or modeled after externpro and we decided that a clean start and a more general location was warranted

File: Findovsrpro.cmake [hunk 7]
```diff
@@ -4,6 +4,8 @@
 # should match xpGetCompilerPrefix in externpro's xpfunmac.cmake
 # NOTE: wanted to use externpro version, but chicken-egg problem
 function(getCompilerPrefix _ret)
+  set(options GCC_TWO_VER)
+  cmake_parse_arguments(x "${options}" "" "" ${ARGN})
```
Comment (smanders) [inline Findovsrpro.cmake:8]: this file shouldn't even exist in this repo -- it should be generated at build-time (or actually cmake-time) -- there is a template in externpro https://github.com/smanders/externpro/blob/18.01.1/modules/Findscript.cmake.in that creates this file and I can't think of any good reason to commit it to the repo

File: projects/qwt.cmake [hunk 8]
```diff
@@ -30,15 +30,15 @@ function(download_qwt)
...
+  ipDownload(${PRO_QWT})
 endfunction()
 ########################################
 function(patch_qwt)
   if(NOT (XP_DEFAULT OR XP_PRO_QWT))
     return()
   endif()
 
-  xpPatch(${PRO_QWT})
+  ipPatch(${PRO_QWT})
```
Comment (smanders) [inline projects/qwt.cmake:41]: I'll submit a pull request to get rid of all the `ip` function calls... they should only be used internally by externpro https://github.com/smanders/externpro/blob/18.01.1/modules/xpfunmac.cmake#L4 and there shouldn't be any reason for overriding the mkpatch, download, and patch steps any more

**Comment by bnhhughes** [comment]
@smanders, those are compelling reasons, I didn't realize that much was added to the latest externpro - I just didn't see us using anything new from the latest version.  It would make sense to take advantage of FFmpeg and OpenH264 in there.

**Comment by swflb** [comment]
@smanders thanks for all the suggestions. I am working to implement them.

File: projects/README.md (hunk 2 above)
Comment (smanders) [reply projects/README.md]: the `projects/README.md` file is generated at cmake time (not build time) -- but only if the `XP_STEP` is *not* `STREQUAL "build"` -- you can see that every step calls `xpMarkdownReadmeFinalize()` *except* `build` in the `proExecuteStep()` macro https://github.com/smanders/externpro/blob/18.01.1/modules/macpro.cmake#L196-L218

**Comment by smanders** [comment]
@swflb I guess you didn't see my comment that I would submit a pull request to get rid of all the `ip` function calls... I can see that you're making quite a few changes to your master branch https://github.com/swflb/ovsrpro/commits/master so I'll wait until the dust settles
